src/tasks/index.ts

```diff
@@ -0,0 +1,176 @@
+		{
+			modelName: apiRoot,
+			apiRoot,
+			modelText,
+			customServerCode: exports,
+			initSql,
+		},
+	],
+};
+
+let worker: Worker | null = null;
+export async function setup(): Promise<void> {
+	// Async task functionality is only supported on Postgres
+	if (sbvrUtils.db.engine !== 'postgres') {
+		console.warn('Skipping task setup as database not supported');
+		return;
+	}
+
+	const client = sbvrUtils.api[apiRoot];
+	worker = new Worker(client);
+
+	// Add resource hooks
+	addPureHook('POST', apiRoot, 'task', {
+		POSTPARSE: async ({ req, request }) => {
+			// Set the actor
+			request.values.is_created_by__actor =
+				req.user?.actor ?? req.apiKey?.actor;
+			if (request.values.is_created_by__actor == null) {
+				throw new BadRequestError(
+					'Creating tasks with missing actor on req is not allowed',
+				);
+			}
+
+			// Set defaults
+			request.values.status = 'queued';
+			request.values.attempt_count = 0;
+			request.values.attempt_limit ??= 1;
+
+			// Set scheduled start time using cron expression if provided
+			if (
+				request.values.is_scheduled_with__cron_expression != null &&
+				request.values.is_scheduled_to_execute_on__time == null
+			) {
+				try {
+					request.values.is_scheduled_to_execute_on__time = cronParser
+						.parseExpression(request.values.is_scheduled_with__cron_expression)
+						.next()
+						.toDate()
+						.toISOString();
+				} catch {
+					throw new BadRequestError(
+						`Invalid cron expression: ${request.values.is_scheduled_with__cron_expression}`,
+					);
+				}
+			}
+
+			// Assert that the provided start time is far enough in the future
+			if (request.values.is_scheduled_to_execute_on__time != null) {
+				const now = new Date(Date.now() + tasksEnv.queueIntervalMS);
+				const startTime = new Date(
+					request.values.is_scheduled_to_execute_on__time,
+				);
+				if (startTime < now) {
+					throw new BadRequestError(
+						`Task scheduled start time must be greater than ${tasksEnv.queueIntervalMS} milliseconds in the future`,
+					);
+				}
+			}
+
+			// Assert that the requested handler exists
+			const handlerName = request.values.is_executed_by__handler;
+			if (handlerName == null) {
+				throw new BadRequestError(`Must specify a task handler to execute`);
+			}
+			const handler = worker?.handlers[handlerName];
+			if (handler == null) {
+				throw new BadRequestError(
+					`No task handler with name '${handlerName}' registered`,
+				);
+			}
+
+			// Assert that the provided parameter set is valid
+			if (handler.validate != null) {
+				if (!handler.validate(request.values.is_executed_with__parameter_set)) {
+					throw new BadRequestError(
+						`Invalid parameter set: ${ajv.errorsText(handler.validate.errors)}`,
+					);
+				}
+			}
+		},
+	});
+	addPureHook('all', apiRoot, 'task', {
+		// Convert bigints to strings in responses
+		PRERESPOND: async ({ response }) => {
+			if (typeof response.body === 'object') {
+				convertBigIntsToStrings(response.body);
```

Comment:
The accidental major sbvr-types bump has been reverted so this should no longer be necessary and we also plan to change sbvr-types so it will not do the automatic conversion to bigint since it shouldn't be automatically converting to JSON incompatible types in fetchProcessing